public/js/collapse_preferences.js

```diff
@@ -1,13 +1,20 @@
 export default class CollapsePreferences {
-    constructor(component) {
+    constructor(component, reactFunction = false) {
```

Comment:
This change feels wrong, you are essentially changing the function interface by passing in `reactFunction = true`. We don't use `CollapsePreferences` in many places, would it be fairly easy to update the other components to `useState` and `useEffect` so that `CollapsePreferences` interface remains the same?

Reply:
I did use the `collapsePreferences` function in a React functional component before (in the `ReportHeader`). Now, the collapse preferences have reverted to their initial state.